acoular/calib.py

```diff
@@ -76,3 +88,29 @@ def import_data(self):
             data.append(float(element.getAttribute('factor')))
         self.data = array(data, 'd')
         self.num_mics = self.data.shape[0]
+
+    def result(self, num):
+        """Python generator that processes the source data and yields the time-signal block-wise.
+
+        This method needs to be implemented by the derived classes.
+
+        Parameters
+        ----------
+        num : int
+            This parameter defines the size of the blocks to be yielded
+            (i.e. the number of samples per block)
+
+        Yields
+        ------
+        numpy.ndarray
+            Two-dimensional output data block of shape (num, numchannels)
+        """
+        for block in self.source.result(num):
+            if self.data.shape[0] == self.num_mics:
+                if self.num_mics == self.numchannels:
+                    yield block * self.data[newaxis]
```

Comment:
This line will only work in the time domain. If the source is of type `SpectraGenerator`, the data will is of shape `(num, numchannels*numfreq)` or even `(num, numchannels*numchannels*numfreq)`. A unittest that ensures the correct behaviour would be beneficial for this class.